File: tests/perf/test-array-push.lua
local function test()
    for i=1,1e5 do
        local arr = {}
        for j=1,10 do
            table.insert(arr, 'foo'); table.insert(arr, 'bar')
            table.insert(arr, 'foo'); table.insert(arr, 'bar')
            table.insert(arr, 'foo'); table.insert(arr, 'bar')
            table.insert(arr, 'foo'); table.insert(arr, 'bar')
            table.insert(arr, 'foo'); table.insert(arr, 'bar')
        end
    end
end

test()

File: tests/perf/test-array-push.py
def test():
    i = 0
    while i < 1e5:
        arr = []
        j = 0
        while j < 10:
            arr.append('foo'); arr.append('bar')
            arr.append('foo'); arr.append('bar')
            arr.append('foo'); arr.append('bar')
            arr.append('foo'); arr.append('bar')
            arr.append('foo'); arr.append('bar')
            j += 1
        i += 1

test()
